src/errors.ts

```diff
@@ -19,43 +19,101 @@
 
 import * as http from 'http'
 import { DiagnosticResult } from './types'
+import { redactObject } from './security'
+
+export interface ErrorOptions {
+  redactDiagnostics: boolean
+  redactConnection: boolean
+  additionalRedactionKeys: string[]
+}
 
 export class ElasticsearchClientError extends Error {
-  constructor (message: string) {
+  options: ErrorOptions
+  constructor (message: string, options?: ErrorOptions) {
     super(message)
     this.name = 'ElasticsearchClientError'
+
+    this.options = {
+      redactDiagnostics: true,
+      redactConnection: false,
+      additionalRedactionKeys: []
+    }
+
+    if (isObject(options)) {
+      this.options = { ...this.options, ...options }
+    }
   }
 }
 
 export class TimeoutError extends ElasticsearchClientError {
   meta?: DiagnosticResult
-  constructor (message: string, meta?: DiagnosticResult) {
-    super(message)
+  constructor (message: string, meta?: DiagnosticResult, options?: ErrorOptions) {
+    if (isObject(meta) && !isObject(options)) {
+      throw new ConfigurationError("If 'meta' is provided, 'options' must also be provided")
+    }
```

Comment:
This enforcement exposed a few surprise places in `elasticsearch-js` that import and use these `Error` classes! Success!

Given that, I will likely need to bump this library to 9.0.0 when releasing, since it does technically break its current contract with the client.

Reply:
After talking with the team, I may pull this out, so that the contract doesn't break and we can keep the transport at 8.x. A slightly less ideal position, but only for people using the transport library directly, as the client itself will always provide these options.